onnxconverter_common/auto_mixed_precision_model_path.py

```diff
@@ -0,0 +1,250 @@
+    location = "fp16_tensor.data"
+
+
+    # Could also use rtol/atol attributes directly instead of this
+    def validate(res1, res2):
+        for r1, r2 in zip(res1, res2):
+            if not np.allclose(r1, r2, rtol=0.01, atol=0.001):
+                return False
+        return True
+
+    auto_convert_mixed_precision_model_path(
+        source_model_path, test_data,
+        target_model_path, location=None,
+        validate_fn=None, rtol=None, atol=None,
+        keep_io_types=True, providers=None)
+
+You don't need to call onnx.save_model() in the customer code.
+The target fp16 model and its external data will be saved as %target_model_path and %location.
+Please note that keep_io_types=True, so the test_data can be used for both fp32 and fp16 models during the convertion.
+
+"""
+
+import copy
+import numpy as np
+import onnxruntime as ort
+import onnx
+import os
+import uuid
+from onnxconverter_common import float16
+from onnx import shape_inference
+from .auto_mixed_precision import SegmentList
+
+
+def auto_convert_mixed_precision_model_path(source_model_path, input_feed,
+                                            target_model_path, location=None,
+                                            validate_fn=None, rtol=None, atol=None,
+                                            keep_io_types=False, providers=None):
+    """
+    Automatically converts a model to mixed precision, excluding the minimum number of nodes required to
+    ensure valudate_fn returns True and/or results are equal according to rtol/atol
+    this version support model_path as input, which the model could > 2GB
+    """
+
+    if not isinstance(source_model_path, str):
+        raise TypeError('auto_convert_mixed_precision_model_path only accepts model Path (String),'
+                        'you can use auto_convert_mixed_precision for the ModelProto.')
+
+    if rtol is None and atol is not None:
+        rtol = 1e-5
+
+    if atol is None and rtol is not None:
+        atol = 1e-8
+
+    if rtol is None and validate_fn is None:
+        raise ValueError("Argument `validate_fn` and `rtol` cannot both be `None`.")
+
+    if location is None:
+        location = "fp16_tensor.data"
+
+    tmp_model32_path, tmp_model32_tensor_name = generate_temp_filename(target_model_path)
+
+    kwargs = {
+        "tmp_model32_path": tmp_model32_path,
+        "tmp_model32_tensor_name": tmp_model32_tensor_name,
+        "source_model_path": source_model_path,
+        "input_feed": input_feed,
+        "target_model_path": target_model_path,
+        "location": location,
+        "validate_fn": validate_fn,
+        "rtol": rtol,
+        "atol": atol,
+        "keep_io_types": keep_io_types,
+        "providers": providers
+        }
+
+    print("copy source model to temp folder, change to external data, then check.")
+    model_32, output_32 = _copy_fp32_model(**kwargs)
```

Comment:
Suggest changing the name from _copy_fp32_model() to _adjust_and_inference_source_model()